### ISSUE TITLE
Socket observable swallows exceptions

### ISSUE THREAD

**Issue by Frizi** [rethinkdb/horizon#536]
When an exception happen to be thrown from inside `.subscribe` on any watched query, the unsubscription is being made by rxjs, but the error remains silent.
I found this particular line in my callstack after trapping on caught exceptons
https://github.com/rethinkdb/horizon/blob/next/client/src/socket.js#L236

Btw, what do you think about using the socket subject implementation from rxjs?
https://github.com/ReactiveX/rxjs/blob/master/src/observable/dom/WebSocketSubject.ts
I haven't found any official documentation for that though.

**Comment by danielmewes** [comment]
Thanks for opening an issue @Frizi .

This sounds like a duplicate of https://github.com/rethinkdb/horizon/issues/500

I'm not quite sure if the line in socket.js that you found is actually relevant, as it seems like `subscribe` always discards exceptions. Pinging @deontologician .

**Comment by deontologician** [comment]
I saw this recently, I believe the error suppression was to avoid situations where we would error an observable that was already closed. That was probably not the right way to handle it.

Rxjs 5's WebSocketSubject is based off the same original implementation I based our socket off of (https://github.com/blesh/RxSocketSubject) which is unmaintained, so we couldn't add it as a dependency. In addition, I integrated some of our serialization/request handling logic into the socket layer, since I was writing it from scratch anyway. Using the rxjs 5 socketsubject would probably reduce some of the file-size, but we'd still need that socket class to handle things like the handshake and request_id handling. It's probably a good idea though
